lib/tree.rb

```diff
@@ -16,47 +16,141 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  # Time Complexity: O(n)
+  # Space Complexity: O(n)
+  def add(key, value=nil)
```

Comment:
👍 The time/space complexity is O(log n) if the tree is balanced and O(n) if it is unbalanced.